src/clj/harja/palvelin/raportointi/raportit/vastaanottotarkastus.clj

```diff
@@ -263,49 +294,137 @@
+
+        ;; Groupataan ja järjestellään rivit hallintayksiköittäin, kuten pk-luokka taulukossakin
+        elyttain-jaoteltu (sort-by first (group-by :hallintayksikko_nimi urakkarivit))
+
+        formatoi-elyt-fn (fn [ely]
+                           (let [elyrivi {:otsikko (str (:elynumero (first (second ely))) " " (first ely))}
+                                 urakat-lista (mapv
+                                                (fn [rivi]
+                                                  (pkluokka-yotyo-rivi rivi false false))
+                                                (second ely))
+                                 ely-yhteensa (reduce fn-laske-yhteen
+                                                (yhtrivi-pohja (str (first ely) " yht. / KA%"))
+                                                (second ely))]
+
+                             (vec (flatten [elyrivi (into [] urakat-lista) (pkluokka-yotyo-rivi ely-yhteensa true true)]))))
+
+        valtukunnallinen-rivi (pkluokka-yotyo-rivi valtakunnallisesti-yhteensa true true)]
+
+    [:taulukko {:tyhja (when (empty? urakkarivit) "Ei kohteita.")
+                :sheet-nimi "Yötyö / PK-luokka"
+                :rivi-ennen [{:sarakkeita 1 :leveys 10}
+                             {:teksti "PK1 yötyö"
+                              :sarakkeita 2
+                              :luokka "paallystys-tausta-tumma"
+                              :tummenna-teksti? true
+                              :tasaa :keskita}
+                             {:teksti "PK2 yötyö"
+                              :sarakkeita 2
+                              ;;:luokka "paallystys-tausta-tumma"
+                              :tummenna-teksti? true
+                              :tasaa :keskita}
+                             {:teksti "PK3 yötyö"
+                              :sarakkeita 2
+                              :luokka "paallystys-tausta-tumma"
+                              :tummenna-teksti? true
+                              :tasaa :keskita}
+                             {:teksti "Ei tiedossa yötyö"
+                              :sarakkeita 2
+                              ;;:luokka "paallystys-tausta-tumma"
+                              :tummenna-teksti? true
+                              :tasaa :keskita}]}
+     ;; Otsikot
+     (concat
+       [{:otsikko "Urakka" :leveys 10}
+        {:otsikko "Pituus (km)" :leveys 2 :fmt :numero :tasaa :oikea}
+        {:otsikko "Prosenttiosuus" :leveys 2 :fmt :prosentti :tasaa :vasen}
+        {:otsikko "Pituus (km)" :leveys 2 :fmt :numero :tasaa :oikea}
+        {:otsikko "Prosenttiosuus" :leveys 2 :fmt :prosentti :tasaa :vasen}
+        {:otsikko "Pituus (km)" :leveys 2 :fmt :numero :tasaa :oikea}
+        {:otsikko "Prosenttiosuus" :leveys 2 :fmt :prosentti :tasaa :vasen}
+        {:otsikko "Pituus (km)" :leveys 2 :fmt :numero :tasaa :oikea}
+        {:otsikko "Prosenttiosuus" :leveys 2 :fmt :prosentti :tasaa :vasen}])
```

Comment:
Tee otsikoista figman mukaisia (ks. yllä liitetty kuva) : 

```clojure
(concat
       [{:otsikko "Urakka" :leveys 10}
        {:otsikko "Pituus (km)" :leveys 2 :fmt :numero :tasaa :oikea :otsikkorivi-luokka "paallystys-tausta-tumma"}
        {:otsikko "Prosenttiosuus" :leveys 2 :fmt :prosentti :tasaa :vasen :otsikkorivi-luokka "paallystys-tausta-tumma"}
        {:otsikko "Pituus (km)" :leveys 2 :fmt :numero :tasaa :oikea}
        {:otsikko "Prosenttiosuus" :leveys 2 :fmt :prosentti :tasaa :vasen}
        {:otsikko "Pituus (km)" :leveys 2 :fmt :numero :tasaa :oikea :otsikkorivi-luokka "paallystys-tausta-tumma"}
        {:otsikko "Prosenttiosuus" :leveys 2 :fmt :prosentti :tasaa :vasen :otsikkorivi-luokka "paallystys-tausta-tumma"}
        {:otsikko "Pituus (km)" :leveys 2 :fmt :numero :tasaa :oikea}
        {:otsikko "Prosenttiosuus" :leveys 2 :fmt :prosentti :tasaa :vasen}])
```

Reply:
Kaikki ui muutokset tehty kuten figmassa. Kiitos erittäin hyvästä palautteesta!